Fix parsing by using libxml; add title, alt text, and after comic; and other improvements

Miciah opened the pull request:
This pull request comprises logical commits that form a progression.  I hope you will like all the changes, but if you prefer a subset, I will be happy to submit individual pull requests for any commits you will accept.

#### Fix parsing by using libxml

Fix sakshamsharma/emacs-smbc#3.

* `smbc.el` (`smbc-parse-html`): Replace regexps and string manipulation with libxml and DOM manipulation.  Because the `libxml-parse-html-region` function requires a buffer as input, replace the string argument to `smbc-parse-html` with a buffer argument.  Because SMBC no longer consistently uses sequential integers for image ids, parse the links to the previous and next comics and store using the `smbc-previous-url` and `smbc-next-url` variables.
(`smbc-retrieve`): New function that retrieves the given URL or path on smbc-comics.com and returns a buffer containing the content.
(`smbc-get-previous`, `smbc-get-next`): Use `smbc-retrieve` and the `smbc-previous-url` or `smbc-next-url` variable, respectively, or print a message indicating if the variable is null.
(`smbc-get-image-data`, `smbc-get-index-page`): Use `smbc-retrieve`.
(`smbc-get-page-given-id`, `smbc-chomp`): Delete.


#### Add title and alt text

* `smbc.el` (`smbc-parse-html`): Parse the comic title and alt text and store them in the `smbc-current-title` and `smbc-current-alt` variables.
(`smbc-display-in-buffer`): Insert the title and alt text.


#### Leave point at start of buffer

* `smbc.el` (`smbc-display-in-buffer`): Use `save-excursion` to avoid leaving the point at the bottom of the buffer, which could cause the comic to be scrolled out of view.


#### Fix `smbc-get-image-from-image-id`

Commit 6ee75f593c5ff32b6ced272081b02a3bfd52ba4b deleted the `smbc-get-image` function, which broke `smbc-get-image-from-image-id`.

* `smbc.el` (`smbc-get-image-from-image-id`): Rewrite not to use the `smbc-get-image` function (which no longer exists).


#### Show the after comic.

* `smbc.el` (`smbc-parse-html`): Parse the after comic image URL and store it in the `smbc-after-image-url` variable.
(`smbc-display-in-buffer`): Retrieve and insert the after comic, if any.

sakshamsharma commented:
This looks pretty good ! Just tested, seems to work as expected. Thanks a lot for this super neat PR, and happy holidays! 🎉  IIRC it should get updated on melpa once this is merged.

Just a minor change left, there's a TODO at the beginning of the file which I'd added to remind myself to shift to `xml`, it should be removed now.

Side note:
There were some issues with `libpng-config` on my Gentoo and thus I never got around to fixing this (Couldn't compile emacs with png support), but I spent half an hour fixing it today, to test this. Works ! 🎉 

Also:
fixes #3

Miciah commented:
Thanks, and happy holidays! I updated the first commit to delete the TODO and added mention of sakshamsharma/emacs-smbc#3 to the PR description.  Does that look good?

sakshamsharma commented:
Yes, looks good. Thanks again! Merging.
